src/main/java/uk/gov/pay/directdebit/payments/services/PaymentRequestService.java

```diff
@@ -0,0 +1,147 @@
+package uk.gov.pay.directdebit.payments.services;
+
+import com.google.common.collect.ImmutableMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.directdebit.app.config.DirectDebitConfig;
+import uk.gov.pay.directdebit.app.config.LinksConfig;
+import uk.gov.pay.directdebit.payments.api.PaymentRequestResponse;
+import uk.gov.pay.directdebit.payments.dao.PaymentRequestDao;
+import uk.gov.pay.directdebit.payments.dao.PaymentRequestEventDao;
+import uk.gov.pay.directdebit.payments.dao.TokenDao;
+import uk.gov.pay.directdebit.payments.dao.TransactionDao;
+import uk.gov.pay.directdebit.payments.exception.ChargeNotFoundException;
+import uk.gov.pay.directdebit.payments.model.*;
+
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import java.net.URI;
+import java.time.ZonedDateTime;
+import java.util.*;
+
+import static javax.ws.rs.HttpMethod.GET;
+import static javax.ws.rs.HttpMethod.POST;
+import static javax.ws.rs.core.MediaType.APPLICATION_FORM_URLENCODED;
+import static uk.gov.pay.directdebit.common.resources.V1ApiPaths.CHARGE_API_PATH;
+
+public class PaymentRequestService {
+    private static final Logger logger = LoggerFactory.getLogger(PaymentRequestService.class);
+
+    private final LinksConfig linksConfig;
+    private final TokenDao tokenDao;
+    private final PaymentRequestDao paymentRequestDao;
+    private final PaymentRequestEventDao paymentRequestEventDao;
+    private final TransactionDao transactionDao;
+
+    public PaymentRequestService(DirectDebitConfig config, PaymentRequestDao paymentRequestDao, TokenDao tokenDao, PaymentRequestEventDao paymentRequestEventDao, TransactionDao transactionDao) {
+        this.paymentRequestDao = paymentRequestDao;
+        this.paymentRequestEventDao = paymentRequestEventDao;
+        this.tokenDao = tokenDao;
+        this.transactionDao = transactionDao;
+        this.linksConfig = config.getLinks();
+    }
+    private Map<String, Object> createLink(String rel, String method, URI href) {
+        return ImmutableMap.of(
+                "rel", rel,
+                "method", method,
+                "href", href
+        );
+    }
+
+    private Map<String, Object> createLink(String rel, String method, URI href, String type, Map<String, Object> params) {
+        return ImmutableMap.of(
+                "rel", rel,
+                "method", method,
+                "href", href,
+                "type", type,
+                "params", params
+        );
+    }
+
+    public PaymentRequestResponse populateResponseWith(PaymentRequest paymentRequest, Transaction charge, UriInfo uriInfo) {
+        String paymentRequestExternalId = paymentRequest.getExternalId();
+        List<Map<String, Object>> dataLinks = new ArrayList<>();
+        dataLinks.add(createLink("self", GET, selfUriFor(uriInfo, paymentRequest.getGatewayAccountId(), paymentRequestExternalId)));
+        if (!charge.getState().toExternal().isFinished()) {
```

Comment:
So next_url is being used only for getting one single charge but not a list, if this is copied from connector we could change that for direct debit connector